packages/semcom-components/lib/components/input.component.ts

```diff
@@ -90,35 +123,48 @@ export default class InputComponent extends BaseComponent {
    */
   handleClickSubmit() {
 
-    if (!this.entry) {
+    let location = this.entry;
 
-      throw new Error('Argument this.entry should be set.');
+    if (this.fileName?.value) {
+
+      location = this.fileName.value;
 
     }
 
-    if (!this.input) {
+    if (!this.content) {
 
       throw new Error('Argument this.inputField should be set.');
 
     }
 
-    this.input.disabled = true;
+    this.content.disabled = true;
+    this.fileName.disabled = true;
     this.button.disabled = true;
+    this.showAlert = false;
+
+    const data = [ new Quad(new NamedNode(location), new NamedNode('https://digita.ai/voc/foo/bar'), new Literal(this.content.value)) ];
+    this.writeData(location, data);
+
+  }
 
-    const data = [ new Quad(new NamedNode(this.entry), new NamedNode('https://digita.ai/voc/foo/bar'), new Literal(this.input.value)) ];
+  toggleAlert() {
```

Comment:
A toggle function should really switch both ways. Here it is only used in one place, afaik, so maybe just use `() => this.showAlert = false` there in the template.